include/mbgl/storage/default_file_source.hpp

```diff
@@ -2,9 +2,135 @@
 #define MBGL_STORAGE_DEFAULT_FILE_SOURCE
 
 #include <mbgl/storage/file_source.hpp>
+#include <mbgl/util/geo.hpp>
+
+#include <vector>
 
 namespace mbgl {
 
+/*
+ * An offline region defined by a style URL, geographic bounding box, zoom range, and
+ * device pixel ratio.
+ */
+class OfflineTilePyramidRegionDefinition {
+public:
+    OfflineTilePyramidRegionDefinition(const std::string&, const LatLngBounds&, double, double, float);
```

Comment:
It would be good to have a constructor where minZoom and maxZoom is optional since they can be automatically inferred for all sources from the style.